Mathlib/Analysis/CStarAlgebra/Hom.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Order
+
+/-! # Properties of C⋆-algebra homomorphisms
+
+Here we collect properties of C⋆-algebra homomorphisms.
+
+## Main declarations
+
++ `NonUnitalStarAlgHom.norm_map`: A non-unital star algebra monomorphism of complex C⋆-algebras
+  is isometric.
+-/
+
+namespace NonUnitalStarAlgHom
+
+variable {F A B : Type*}
+variable [NonUnitalNormedRing A] [CompleteSpace A] [StarRing A] [CStarRing A]
+variable [NormedSpace ℂ A] [IsScalarTower ℂ A A] [SMulCommClass ℂ A A] [StarModule ℂ A]
+variable [NonUnitalNormedRing B] [CompleteSpace B] [StarRing B] [CStarRing B]
+variable [NormedSpace ℂ B] [IsScalarTower ℂ B B] [SMulCommClass ℂ B B] [StarModule ℂ B]
+variable [FunLike F A B] [NonUnitalAlgHomClass F ℂ A B] [NonUnitalStarAlgHomClass F ℂ A B]
+
+open CStarAlgebra Unitization in
+/-- A non-unital star algebra monomorphism of complex C⋆-algebras is isometric. -/
+lemma norm_map (φ : F) (hφ : Function.Injective φ) (a : A) : ‖φ a‖ = ‖a‖ := by
+  /- by passing to the unitization is functorial, and it is an isometric embedding, we may assume
+  that `φ` is a unital star algebra monomorphism and that `A` and `B` are unital C⋆-algebras. -/
+  suffices ∀ {ψ : Unitization ℂ A →⋆ₐ[ℂ] Unitization ℂ B} (_ : Function.Injective ψ)
+      (a : Unitization ℂ A), ‖ψ a‖ = ‖a‖ by
+    simpa [norm_inr] using this (starMap_injective (φ := (φ : A →⋆ₙₐ[ℂ] B)) hφ) a
+  intro ψ hψ a
+  -- to show `‖ψ a‖ = ‖a‖`, by the C⋆-property it suffices to show `‖ψ (star a * a‖ = ‖star a * a‖`.
+  rw [← sq_eq_sq (by positivity) (by positivity)]
+  simp only [sq, ← CStarRing.norm_star_mul_self, ← map_star, ← map_mul]
+  /- since `star a * a` is selfadjoint, it suffices to show that it has the same `ℝ`-spectrum as
+  `ψ (star a * a)`, since the spectral radius coincides with the norm. -/
+  suffices ∀ a, IsSelfAdjoint a → spectrum ℝ (ψ a) = spectrum ℝ a from
```

Comment:
It feels like that statement deserves its own lemma, no?

Reply:
Perhaps, yes. I'll do it later tonight.